Automattic/jetpack
# JS Components: Convert DecorativeCard component to TypeScript

#### Changes proposed in this Pull Request:
* This PR converts `DecorativeCard` component to TypeScript and cleans up the propTypes.
* It also converts the `DecorativeCard` stories to TS

#### Other information:


#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the stories of `DecorativeCard` component work as expected
* Check the consumers of `DecorativeCard` and related components and confirm that they work as expected
* Check for any regression


<img width="1125" alt="Screenshot 2022-07-13 at 12 00 02 PM" src="https://user-images.githubusercontent.com/18226415/178665868-989a8579-bd33-412a-8287-af55a455de05.png">
<img width="1113" alt="Screenshot 2022-07-13 at 12 00 11 PM" src="https://user-images.githubusercontent.com/18226415/178665883-cf81c467-f28c-4b91-b483-acfa980ee9e1.png">


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202276314263997
